metal3-dev/run.sh

```diff
@@ -68,7 +68,10 @@ export IRONIC_ENDPOINT=http://172.22.0.3:6385/v1/
 export IRONIC_INSPECTOR_ENDPOINT=http://172.22.0.3:5050/v1/
 
 # Wait for the ironic service to be available
-wait_for_json ironic "$IRONIC_ENDPOINT" 300 \
+export MASTER0=master-0.$CLUSTER_DOMAIN
+export IRONIC_ENDPOINT_PUBLIC=$(oc get nodes -o json | yq -r '.items[] | select(.metadata.name == env.MASTER0) | .status.addresses[] | select(.type == "InternalIP") | .address')
```

Comment:
Why does this look at master-0? There's no guarantee where Ironic gets scheduled

Reply:
That's right, it was just a matter of convenience in the dev env (quick), in absence of a better ingress to check ironic status from outside the cluster.

Reply:
I think we could read the node name off of the scheduled pod, too. But we also know that the pod uses a VIP on the provisioning network, and that's going to be more reliable because it doesn't matter which host the pod is on. Is that information available in the pod definition? Or do we have to re-compute it?

Reply:
Yea it should be in the environment variables on the static-ip-manager container in the metal3 pod.

It's also just the 3rd host in the provisioning network, by default. I doubt any devs are altering that behavior. You can see how we calculate the PROVISIONING_HOST_IP in common.sh with python's ipaddress package.

Reply:
I checked and in the pod there's 
```
{
                        "name": "IRONIC_ENDPOINT",
                        "value": "http://[fd00:1101::3]:6385/v1/"
 }
```

Reply:
If the value is available from a variable in common, we could just use that variable here in this script. It sources common.sh already.

Reply:
Ok I'll check it thanks

Reply:
The address on which Ironic is listening is not available in any of the envvars in common.sh. The `PROVISIONING_HOST_IP` ends in a `::1` whereas the correct value ends in `::3`. 

Here is the relevant line from `common.sh`:

```bash
export PROVISIONING_HOST_IP=${PROVISIONING_HOST_IP:-$(python -c "import ipaddress; print(next(ipaddress.ip_network(u\"$PROVISIONING_NETWORK\").hosts()))")}
```

Reply:
#1040